packages/frontend/src/components/VisualizationConfigs/TableConfigPanel/ConditionalFormatting.tsx

```diff
@@ -49,82 +66,181 @@ const ConditionalFormatting: FC<ConditionalFormattingProps> = ({
+                        newConfigType,
+                        'Unknown config type',
+                    );
+            }
+        },
+        [handleChange, config],
+    );
+
+    const handleAddRule = useCallback(() => {
         setIsAddingRule(true);
-        handleChange(
-            produce(config, (draft) => {
-                draft.rules.push(createConditionalFormatingRule());
-            }),
-        );
-    };
-
-    const handleRemoveRule = (index: number) => {
-        handleChange(
-            produce(config, (draft) => {
-                draft.rules.splice(index, 1);
-            }),
-        );
-    };
-
-    const handleChangeRuleOperator = (
-        index: number,
-        newOperator: ConditionalOperator,
-    ) => {
-        handleChange(
-            produce(config, (draft) => {
-                draft.rules[index].operator = newOperator;
-            }),
-        );
-    };
-
-    const handleChangeRule = (
-        index: number,
-        newRule: ConditionalFormattingRuleT,
-    ) => {
-        handleChange(
-            produce(config, (draft) => {
-                draft.rules[index] = newRule;
-                // FIXME: check if we can fix this problem in number input
-                draft.rules[index].values = draft.rules[index].values.map((v) =>
-                    Number(v),
+
+        if (isConditionalFormattingConfigWithSingleColor(config)) {
+            handleChange(
+                produce(config, (draft) => {
+                    draft.rules.push(createConditionalFormatingRule());
+                }),
+            );
+        }
+    }, [handleChange, config]);
+
+    const handleRemoveRule = useCallback(
+        (index: number) => {
+            if (isConditionalFormattingConfigWithSingleColor(config)) {
+                handleChange(
+                    produce(config, (draft) => {
+                        draft.rules.splice(index, 1);
+                    }),
+                );
+            }
+        },
+        [handleChange, config],
+    );
+
+    const handleChangeRuleOperator = useCallback(
+        (index: number, newOperator: ConditionalOperator) => {
+            if (isConditionalFormattingConfigWithSingleColor(config)) {
+                handleChange(
+                    produce(config, (draft) => {
+                        draft.rules[index] = {
+                            ...draft.rules[index],
+                            operator: newOperator,
+                        };
+                    }),
+                );
+            }
+        },
+        [handleChange, config],
+    );
+
+    const handleChangeRule = useCallback(
+        (
+            index: number,
+            newRule: ConditionalFormattingWithConditionalOperator,
+        ) => {
+            if (isConditionalFormattingConfigWithSingleColor(config)) {
+                handleChange(
+                    produce(config, (draft) => {
+                        // FIXME: check if we can fix this problem in number input
```

Comment:
do we still need to fix this ?

Reply:
yep, unfortunately this comes from blueprint input which is in filter components (not part of this change)